### PR TITLE
Ci setup

### PR DESCRIPTION
**Pull request by sarrailhadaptive**
Added a pipeline with basic CI commands.

### CODE REVIEW

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    steps:
+    # Check-out code with GitHub Actions to allow workflow to access our repo
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+        # with: fetch-depth:0 means we are getting the last commit
+          fetch-depth: 0
+        
+      - name: Install JS Dependencies
+        run: yarn install
```
Comment (snony14) [inline .github/workflows/pipeline.yml:53]: I see that there are executions of yarn and npm, why not stick with yarn only or npm only throughout this file?

Comment (sarrailhadaptive) [reply .github/workflows/pipeline.yml:53]: My bad, I was trying the configs in a test project and forgot to switch to yarn in some cases. I'll fix now.

Comment (snony14) [reply .github/workflows/pipeline.yml:53]: Please fix the conflicts in package.json as well and then run yarn

Comment (sarrailhadaptive) [reply .github/workflows/pipeline.yml:53]: Done, changes made in the last push.

File: components/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (algreasley) [inline components/.prettierrc:1]: for no particularly strong reason, I'd like to sync with [RT's prettier conf](https://github.com/AdaptiveConsulting/ReactiveTraderCloud/blob/master/src/client/.prettierrc.yml) (or we change both)
```
printWidth: 80
semi: false
trailingComma: all
```

File: components/src/App.tsx [hunk 3]
```diff
@@ -1,34 +1,30 @@
...
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [count, setCount] = useState(0);
 
   return (
     <div className="App text-3xl font-bold underline">
       <div>
-        <a href="https://vitejs.dev" target="_blank">
+        <a href="https://vitejs.dev" target="_blank" rel="noopener noreferrer">
```
Comment (algreasley) [inline components/src/App.tsx:11]: interesting .. eslint fix did this?

File: components/package.json [hunk 4]
```diff
@@ -6,7 +6,10 @@
   "scripts": {
     "dev": "vite",
     "build": "vite build",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "lint": "eslint .",
```
Comment (algreasley) [inline components/package.json:10]: may be able to revert all these workspace-level changes if we apply to root `package.json` ?

**Comment by algreasley** [comment]
this is great stuff .. will be good to have the pipeline running for everyone, and running the code checks regularly, in case someone has IDE set up wrong .. dw about the number of comments, I got a bit "detailed" on you there 😄

File: components/src/App.tsx (hunk 3 above)
Comment (sarrailhadaptive) [reply components/src/App.tsx:11]: This was a fix I did myself actually, I learned in a previous project that target='_blank' creates a vulnerability that can be solved by adding the line rel='noopener noreferer'

File: .github/workflows/pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize]
+
+jobs:
+  avoid_reduncy:
+    runs-on: ubuntu-22.04
+    steps:
+    # Avoid redundant builds to prevent running more than 1 build in parallel
+      - name: Cancel Previous Redundant Builds
```
[the remark this answers is not in the document]
Comment (algreasley) [reply .github/workflows/pipeline.yml]: as we just noticed what looks like a "built in" version of this, perhaps we could migrate to that .. but no need for this PR

**Review by algreasley — APPROVED** [review]

**Pull request merged**